spring-data-mongodb/src/test/java/org/springframework/data/mongodb/observability/MongoObservationCommandListenerTests.java

```diff
@@ -73,54 +75,62 @@ void setup() {
 	@Test
 	void commandStartedShouldNotInstrumentWhenAdminDatabase() {
 
-		// when
 		listener.commandStarted(new CommandStartedEvent(null, 0, 0, null, "admin", "", null));
 
-		// then
 		assertThat(meterRegistry).hasNoMetrics();
 	}
 
 	@Test
 	void commandStartedShouldNotInstrumentWhenNoRequestContext() {
 
-		// when
 		listener.commandStarted(new CommandStartedEvent(null, 0, 0, null, "some name", "", null));
 
-		// then
 		assertThat(meterRegistry).hasNoMetrics();
 	}
 
 	@Test
 	void commandStartedShouldNotInstrumentWhenNoParentSampleInRequestContext() {
 
-		// when
 		listener.commandStarted(new CommandStartedEvent(new MapRequestContext(), 0, 0, null, "some name", "", null));
 
-		// then
 		assertThat(meterRegistry).hasMeterWithName("spring.data.mongodb.command.active");
 	}
 
 	@Test // GH-4994
 	void commandStartedShouldAlwaysIncludeCollection() {
 
-		// when
 		listener.commandStarted(new CommandStartedEvent(new MapRequestContext(), 0, 0, null, "some name", "hello", null));
 
-		// then
 		// although command 'hello' is collection-less, metric must have tag "db.mongodb.collection"
 		assertThat(meterRegistry).hasMeterWithNameAndTags(
 				"spring.data.mongodb.command.active",
 				Tags.of("db.mongodb.collection", "none"));
 	}
 
+	@Test // GH-5082
+	void reactiveContextCompletesNormally() {
```

Comment:
When I undo the implementation change the test is still green, so it seems not suitable to ensure the proper behavior.